File: techila/examples/python/Tutorial/2_parameters/parameters_dist.py
# Copyright 2012-2013 Techila Technologies Ltd.

# This function contains the function that will be executed during
# computational Jobs. Each Job will multiply the values of the two
# input arguments, 'multip' and 'jobidx'. 'multip' will be same for
# all Jobs, 'jobidx' will receive a different peachvector element.
def parameters_dist(multip, jobidx):

    # Multiply the values of variables 'multip' and 'jobidx'
    result = multip * jobidx

    # Return the value of the 'result' variable from the Job.
    return(result)

File: techila/examples/C/Tutorial/2_parameters/parameters_dist.c
// Copyright 2011-2013 Techila Technologies Ltd.

/* This file contains the Worker Code, which will be compiled in to a binary
when the example is compiled using make. The operations shown in this file
will be executed on the Workers. Input arguments for the program have been
defined in the Local Control Code. */

#include <stdio.h>
#include <stdlib.h>

int main(int argc, char** argv) {
  int multip = atoi(argv[1]); // Read 1st input argument
  int jobidx = atoi(argv[2]);  // Read 2nd input argument

  // Operations corresponding to one iteration in the local program
  int result =  multip * jobidx;

  FILE *fp;
  fp=fopen("techila_output", "w");
  fprintf(fp, "%i\n", result); // Write the result in the output file
  fclose(fp);
  return 0;
}
